### PR TITLE
CLI: Fix compiler globals

### PR DESCRIPTION
**Pull request by josephjclark**
This PR contains a small fix for the CLI/compiler, allowing extra globals to be supported in job code.

Basically when we  compile code and generate an import statement, we sometimes (wrongly) add globals to the list of imports. It's the same problem as back in #54 (back then I was thinking about node globals, today 'm thinking about browser  globals).

I've expanded the list of supported globals to include everything on [MDN's list](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects). I'm sure something else will creep through at some stage.

## Relrases

I hope I've done this right: I've run `changeset version` for the CLI and compiler, which should be released ASAP. There are some unreleased changesets hanging around which should not be released yet.

There's a bit of complication because main is slightly broken - see #98. I've rolled back the compiler's `describe-package` dependency and added a couple of simple integration tests (which fail on main). So the new CLI and compiler releases from here should be stable.

To release, you'll need build everything and just publish `cli` and `compiler`.

### CODE REVIEW

**Comment by josephjclark** [comment]
@stuartc When you're ready tomorrow this is a small fix to the CLI. All versioned up, just needs a release.

File: packages/compiler/src/transforms/add-imports.ts [hunk 1]
```diff
@@ -15,29 +15,80 @@ import type { Transformer } from '../transform';
...
   'clearTimeout',
   'console',
+  'DataView',
   'Date',
+  'decodeURI',
+  'decodeURIComponent',
+  'encodeURI',
+  'encodeURIComponent',
   'Error',
+  'eval',
```
Comment (stuartc) [inline packages/compiler/src/transforms/add-imports.ts:42]: I'm a little concerned about adding `eval` here, obviously I don't want to break any adaptors - but if we allow `eval` we have to ensure it is sufficiently isolated or proxied. Like that vm2 vuln. where the calling context is available via the Error prototype; if you think we should allow it and handle the sandboxing in a more holistic manner in the near future I'm ok with that.

**Review by stuartc — COMMENTED** [review]
@josephjclark this looks good, just would like your thoughts on `eval`.

File: packages/compiler/src/transforms/add-imports.ts (hunk 1 above)
Comment (josephjclark) [reply packages/compiler/src/transforms/add-imports.ts:42]: I'll push a commit in a few minutes with some unit tests, but globals like `Set` are available, while `eval` is disabled and `process` not in scope.

I have raised an issue #104  to review and improve unit testing around this.

Comment (josephjclark) [reply packages/compiler/src/transforms/add-imports.ts:42]: Ok - this is good to go! See the latest commit for a couple of comforting unit tests.

Re release - I suggest we merge this  into main, then release from the `release-next` branch and do it all in one go.

Comment (josephjclark) [reply packages/compiler/src/transforms/add-imports.ts:42]: .. apart from the failing test of course

Comment (josephjclark) [reply packages/compiler/src/transforms/add-imports.ts:42]: Ok, all done!

**Pull request merged**